src/form.js

```diff
@@ -802,7 +802,32 @@ define([
             }
         },
         changeMugType: function (mug, questionType) {
-            this.mugTypes.changeType(mug, questionType);
+            var _this = this;
+            _this.mugTypes.changeType(mug, questionType);
+            _this.setMugActions(mug);
+        },
+        setMugActions: function(mug) {
+            var _this = this,
+                tree = _this.vellum.data.core.$tree,
+                action_id = "add_choice";
+
+            if (mug.__className === 'Select' || mug.__className === 'MSelect') {
+                tree.jstree(true).add_action(mug.ufid, {
```

Comment:
I think this UI code should not be in the form object. It should probably be moved into core.js near where the question tree UI is configured. A separate, but maybe related, issue is that core.js is getting pretty big and contains a lot of things. Maybe we should move the question tree into its own module (probably not in this PR)?